File: 0324-wiggle-sort-ii/0324-wiggle-sort-ii.cpp
class Solution {
public:
    void wiggleSort(vector<int>& nums) {
        sort(nums.begin(),nums.end()); 
        int n = nums.size(); 
        vector<int>ans(n); 
        int low = 1; 
        int high =  n-1; 

        while(low< n){
            ans[low] = nums[high--] ;   
            low+=2;  
        }
        low = 0; 
        while( low<n){
            ans[low] = nums[high--];
            low+=2;  
        }
        for( int i = 0; i< n ; i++){
            nums[i] = ans[i];
        }
        
    }
};

File: 0324-wiggle-sort-ii/0324-wiggle-sort-ii.java
class Solution {
    public void wiggleSort(int[] nums) {
        int n = nums.length;
        int[] ans = new int[n];
        Arrays.sort(nums);
        
        int j = n - 1;
        int i = 1;

      
        while (i < n) {
            ans[i] = nums[j--];
            i += 2;
        }

        i = 0;
      
        while (i < n) {
            ans[i] = nums[j--];
            i += 2;
        }

        
        for (i = 0; i < n; i++) {
            nums[i] = ans[i];
        }
    }
}
